### PR TITLE
chore(scripts): fix stable release promote script

### PR DESCRIPTION
**Pull request by mafredri**
This fixes the promote command to use the right payload...

Switches from GH_ to GITHUB_TOKEN as well and adds gh_auth for completeness.

### CODE REVIEW

**Review by stirby — APPROVED** [review]
Glad to see you found the issue. Nice!

Just giving my ack here, should await devs' review.

File: scripts/release_promote_stable.sh [hunk 1]
```diff
@@ -4,6 +4,9 @@ set -euo pipefail
 # shellcheck source=scripts/lib.sh
 source "$(dirname "${BASH_SOURCE[0]}")/lib.sh"
 
+# Make sure GITHUB_TOKEN is set for the release command.
```
Comment (matifali) [inline scripts/release_promote_stable.sh:7]: This only works when `CODER=true` i.e. inside a Coder workspaces.

**Review by dannykopping — APPROVED** [review]

File: scripts/release_promote_stable.sh (hunk 1 above)
Comment (mtojek) [reply scripts/release_promote_stable.sh:7]: Do we suggest adding an extra "echo" to indicate it?

**Review by mtojek — APPROVED** [review]

File: scripts/release_promote_stable.sh (hunk 1 above)
Comment (mafredri) [reply scripts/release_promote_stable.sh:7]: That’s the expected use-case. And if it isn’t set outside a workspace it’ll print a notice to do `gh auth login`. Just realized that won’t work though, but we could update `lib.sh` to fetch the token via `gh auth token` in this case.

Comment (mafredri) [reply scripts/release_promote_stable.sh:7]: daf03f8d70cec593d3b39d4b0245940e5c32ff4e

Comment (mafredri) [reply scripts/release_promote_stable.sh:7]: Perhaps we also shouldn't override GITHUB_TOKEN if it's set? And maybe support GH_TOKEN too (in lib.sh)? Thoughts @matifali?

File: scripts/lib.sh [hunk 2]
```diff
@@ -144,6 +144,8 @@ gh_auth() {
 			GITHUB_TOKEN=$(coder external-auth access-token github)
 			export GITHUB_TOKEN
 		fi
+	elif token="$(gh auth token --hostname github.com 2>/dev/null)"; then
+		export GITHUB_TOKEN=$token
```
Comment (matifali) [inline scripts/lib.sh:148]: 👍🏼

File: scripts/release_promote_stable.sh (hunk 1 above)
Comment (matifali) [reply scripts/release_promote_stable.sh:7]: Yes as far it works and make it more robust. Let's do this. I don't fully understand the difference between `GH_` and `GITHUB_` prefix for tokens.

Comment (matifali) [reply scripts/release_promote_stable.sh:7]: Yes please go ahead. Not blocking you. It looks perfect.

File: scripts/lib.sh [hunk 3]
```diff
@@ -134,18 +134,22 @@ requiredenvs() {
...
 
 gh_auth() {
-	local fail=0
-	if [[ "${CODER:-}" == "true" ]]; then
-		if ! output=$(coder external-auth access-token github 2>&1); then
-			log "ERROR: Could not authenticate with GitHub."
-			log "$output"
-			fail=1
+	if [[ -z ${GITHUB_TOKEN:-} ]]; then
+		if [[ -n ${GH_TOKEN:-} ]]; then
```
Comment (matifali) [inline scripts/lib.sh:138]: We should always get the new token. The token stored in `GITHUB_TOKEN` may be stale, so I opted to refresh this even though it's set.

`coder external-auth access-token github` handles the auto-refreshing automatically.

**Review by matifali — CHANGES_REQUESTED** [review]
When inside a coder workspace, We should always get the new token. The token stored in `GITHUB_TOKEN` may be stale, so I opted to refresh this even though it's set.

**Comment by mafredri** [comment]
> When inside a coder workspace, We should always get the new token. The token stored in `GITHUB_TOKEN` may be stale, so I opted to refresh this even though it's set.

@matifali Why would there be a token stored in `GITHUB_TOKEN` that's stale, and how would that happen? The reason I made it external auth secondary to the env being set is that during yesterdays release we ran into the issue of the Coder external-auth token (i.e. GitHub App) not having all the required permissions, so we had to manually `gh auth login` and use that token.

**Comment by matifali** [comment]
If someone sets the token manually or inject via `coder_env` that would stale for sure. But as we have full control over the dogfood template and there is no way the `GITHUB_TOKEN` can be automatically set so I think it's fine to merge. I am perhaps over thinking.

**Review by matifali — APPROVED** [review]

**Comment by mafredri** [comment]
> If someone sets the token manually or inject via `coder_env` that would stale for sure. But as we have full control over the dogfood template and there is no way the `GITHUB_TOKEN` can be automatically set so I think it's fine to merge. I am perhaps over thinking.

@matifali I'm happy you are over-thinking, wouldn't want to break any use-cases. I think the best solution would be to probe the validity of the token, and if it's expired run the path within the if-statement. I think we can leave that for future enhancement, though.

**Comment by stirby** [comment]
@mafredri, if this resolves all the script issues we encountered, can should merge soon? Colin and I ran through the scripts for patching mainline and would like to do the same for stable tomorrow.

**Pull request merged**